scripts/report_binary_size.js

```diff
@@ -0,0 +1,148 @@
+        'module',
+        'esm2017',
+        'browser',
+        'react-native',
+        'lite',
+        'lite-esm2017'
+    ];
+
+    const packageInfo = JSON.parse(
+        execSync('npx lerna ls --json --scope @firebase/*', { cwd: repoRoot }).toString()
+    );
+
+    for (const package of packageInfo) {
+        const packageJson = require(`${package.location}/package.json`);
+
+        for (const field of fields) {
+            if (packageJson[field]) {
+                const filePath = `${package.location}/${packageJson[field]}`;
+                const { size } = fs.statSync(filePath);
+                reports.push(makeReportObject(packageJson.name, field, size));
+            }
+        }
+    }
+
+    return reports;
+}
+
+function makeReportObject(sdk, type, value) {
+    return {
+        sdk,
+        type,
+        value
+    };
+}
+
+function generateSizeReport() {
+    const reports = [
+        ...generateReportForCDNScripts(),
+        ...generateReportForNPMPackages()
+    ];
+
+    for (const r of reports) {
+        console.log(r.sdk, r.type, r.value);
+    }
+
+    console.log(`Github Action URL: https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${runId}`);
+
+    return {
+        log: `https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${runId}`,
+        metric: "BinarySize",
+        results: reports
+    };
+}
+
+function constructRequestPath() {
+    const repo = process.env.GITHUB_REPOSITORY;
+    const commit = process.env.GITHUB_SHA;
+    let path = `/repos/${repo}/commits/${commit}/reports`;
+    if (process.env.GITHUB_EVENT_NAME === 'pull_request') {
+        const pullRequestNumber = process.env.GITHUB_PULL_REQUEST_NUMBER;
+        const pullRequestBaseSha = process.env.GITHUB_PULL_REQUEST_BASE_SHA;
+        path += `?pull_request=${pullRequestNumber}&base_commit=${pullRequestBaseSha}`;
+    }
+    return path;
+}
+
+function constructRequestOptions(path) {
+    const accessToken = execSync('gcloud auth print-identity-token', { encoding: 'utf8' }).trim();
+    return {
+        path: path,
+        method: 'POST',
+        headers: {
+            'Authorization': `Bearer ${accessToken}`,
+            'Content-Type': 'application/json'
+        }
+    };
+}
+
+function upload(report) {
+    if (!process.env.GITHUB_ACTIONS) {
+        console.log('Metrics upload is only enabled on CI.');
+        return;
+    }
+
+    const path = constructRequestPath();
+    const options = constructRequestOptions(path);
+
+    console.log(`${report.metric} report:`, report);
+    console.log(`Posting to metrics service endpoint: ${path} ...`);
+
+    const request = https.request(METRICS_SERVICE_URL, options, response => {
+        response.setEncoding('utf8');
+        console.log(`Response status code: ${response.statusCode}`);
+        response.on('data', console.log);
+        response.on('end', () => {
+            if (response.statusCode !== 202) {
+                process.exit(1);
```

Comment:
Can you log the status code that's returned and any error message in case there is an error?

Reply:
The status code is logged at line 135.

Line 136 logs the response body, which will be
* in case of success, an id generated by the backend service for the uploaded report
* in case of failure, various reason from unauthenticated to the error message of service internal exceptions